src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -793,8 +845,25 @@
    NULL
 })
 
-.rs.addJsonRpcHandler("get_args", function(name, src)
+.rs.addJsonRpcHandler("get_args", function(name, src, helpHandler)
 {
+   # call custom help handler if provided
+   if (nzchar(helpHandler)) {
+      helpHandlerFunc <- tryCatch(eval(parse(text = helpHandler)),
+                                  error = function(e) NULL)
+      if (!is.function(helpHandlerFunc))
+         return(NULL)
+      help <- helpHandlerFunc("completion", name, src)
+      if (is.null(help))
+         return(NULL)
+      signature <- help$signature
+      if (!is.null(signature)) {
+         paren <- regexpr("\\(", signature)[[1]]
+         signature <- substring(signature, paren)
```

Comment:
Should we be paranoid and double-check that we received a valid signature that contains a parenthesis? (ie, wrap this check in `if (paren != -1) { ... }`)